src/metabase/legacy_mbql/schema/helpers.cljc

```diff
@@ -9,6 +9,36 @@
 
 ;;; --------------------------------------------------- defclause ----------------------------------------------------
 
+(defn mbql-clause?
+  "True if `x` is an MBQL clause (a sequence with a keyword as its first arg). (Since this is used by the code in
+  `normalize` this handles pre-normalized clauses as well.)"
```

Comment:
Good catch... I'm not sure. In practice there's no harm in letting this function be more flexible.

Reply:
I guess I picked the wrong example func. Possibly the point of the versions in the normalize ns is to avoid constantly and unnecessarily calling `maybe-normalize-token` in `is-clause?` for something that should already be normalized. I'll leave the local versions in normalize alone and just update the docstring here.

```clojure
(defn is-clause?
  "If `x` an MBQL clause, and an instance of clauses defined by keyword(s) `k-or-ks`?

    (is-clause? :count [:count 10])        ; -> true
    (is-clause? #{:+ :- :* :/} [:+ 10 20]) ; -> true

  (This is different from the implementation in `mbql.u` because it also supports un-normalized clauses. You shouldn't
  need to use this outside of this namespace.)"
  [k-or-ks x]
  (and
   (mbql-clause? x)
   (let [clause-name (maybe-normalize-token (first x))]
     (if (coll? k-or-ks)
       ((set k-or-ks) clause-name)
       (= k-or-ks clause-name)))))
```